Blazor template tests timing out after https://github.com/dotnet/aspnetcore/pull/45682

```
2023-02-14T17:40:24.1299974Z   Failed Templates.Blazor.Test.BlazorWasmTemplateTest.BlazorWasmHostedPwaTemplateNoHttpsCanCreateBuildPublish [3 s]
2023-02-14T17:40:24.1300326Z   Error Message:
2023-02-14T17:40:24.1300496Z    System.InvalidOperationException : Failed to uninstall previous templates. The template 'reactredux' could still be found.
2023-02-14T17:40:24.1300534Z   Stack Trace:
2023-02-14T17:40:24.1300563Z      at Templates.Test.Helpers.TemplatePackageInstaller.VerifyCannotFindTemplateAsync(ITestOutputHelper output, String templateName) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 151
2023-02-14T17:40:24.1300596Z    at Templates.Test.Helpers.TemplatePackageInstaller.InstallTemplatePackages(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 114
2023-02-14T17:40:24.1300626Z    at Templates.Test.Helpers.TemplatePackageInstaller.EnsureTemplatingEngineInitializedAsync(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 64
2023-02-14T17:40:24.1300655Z    at Templates.Test.Helpers.ProjectFactoryFixture.CreateProject(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/ProjectFactoryFixture.cs:line 29
2023-02-14T17:40:24.1300728Z    at Templates.Blazor.Test.BlazorTemplateTest.CreateBuildPublishAsync(String auth, String[] args, String targetFramework, Boolean serverProject, Boolean onlyCreate) in /_/src/ProjectTemplates/Shared/BlazorTemplateTest.cs:line 49
2023-02-14T17:40:24.1300775Z    at Templates.Blazor.Test.BlazorWasmTemplateTest.BlazorWasmHostedPwaTemplateNoHttpsCanCreateBuildPublish() in /_/src/ProjectTemplates/test/Templates.Blazor.WebAssembly.Tests/BlazorWasmTemplateTest.cs:line 84
2023-02-14T17:40:24.1300804Z --- End of stack trace from previous location ---
2023-02-14T17:40:24.1304531Z   Standard Output Messages:
2023-02-14T17:40:24.1304786Z  | [0.013s] TestLifetime Information: Starting test BlazorWasmHostedPwaTemplateNoHttpsCanCreateBuildPublish at 2023-02-14T17:40:20
2023-02-14T17:40:24.1304820Z  | [0.048s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: ==> /datadisks/disk1/work/A6F508D4/p/dotnet-cli/dotnet new "web" --debug:disable-sdk-templates --debug:custom-hive "/datadisks/disk1/work/A6F508D4/w/A5900923/e/Hives/.templateEngine" [/datadisks/disk1/work/A6F508D4/w/A5900923/e/]
2023-02-14T17:40:24.1304866Z  | [0.527s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'web'.
2023-02-14T17:40:24.1304892Z  | [0.527s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1304916Z  | [0.527s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'web', run:
2023-02-14T17:40:24.1304939Z  | [0.533s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list web
2023-02-14T17:40:24.1304963Z  | [0.533s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.1305065Z  | [0.533s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search web
2023-02-14T17:40:24.1305090Z  | [0.534s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1305176Z  | [0.534s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1305207Z  | [0.534s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.1305304Z  | [1.247s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.1305331Z  | [1.250s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: ==> /datadisks/disk1/work/A6F508D4/p/dotnet-cli/dotnet new "webapp" --debug:disable-sdk-templates --debug:custom-hive "/datadisks/disk1/work/A6F508D4/w/A5900923/e/Hives/.templateEngine" [/datadisks/disk1/work/A6F508D4/w/A5900923/e/]
2023-02-14T17:40:24.1305383Z  | [1.495s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'webapp'.
2023-02-14T17:40:24.1305413Z  | [1.499s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1305521Z  | [1.500s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'webapp', run:
2023-02-14T17:40:24.1305549Z  | [1.503s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list webapp
2023-02-14T17:40:24.1305578Z  | [1.503s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.1305607Z  | [1.504s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search webapp
2023-02-14T17:40:24.1305636Z  | [1.504s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1305661Z  | [1.506s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1305689Z  | [1.506s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.1305717Z  | [1.515s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.1305936Z  | [1.516s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: ==> /datadisks/disk1/work/A6F508D4/p/dotnet-cli/dotnet new "webapi" --debug:disable-sdk-templates --debug:custom-hive "/datadisks/disk1/work/A6F508D4/w/A5900923/e/Hives/.templateEngine" [/datadisks/disk1/work/A6F508D4/w/A5900923/e/]
2023-02-14T17:40:24.1305998Z  | [1.763s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'webapi'.
2023-02-14T17:40:24.1306030Z  | [1.766s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1306058Z  | [1.766s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'webapi', run:
2023-02-14T17:40:24.1306085Z  | [1.768s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list webapi
2023-02-14T17:40:24.1306112Z  | [1.768s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.1306141Z  | [1.768s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search webapi
2023-02-14T17:40:24.1306171Z  | [1.768s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1306201Z  | [1.769s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1306393Z  | [1.770s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.1306441Z  | [1.775s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.1306470Z  | [1.775s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: ==> /datadisks/disk1/work/A6F508D4/p/dotnet-cli/dotnet new "mvc" --debug:disable-sdk-templates --debug:custom-hive "/datadisks/disk1/work/A6F508D4/w/A5900923/e/Hives/.templateEngine" [/datadisks/disk1/work/A6F508D4/w/A5900923/e/]
2023-02-14T17:40:24.1306514Z  | [2.038s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'mvc'.
2023-02-14T17:40:24.1306538Z  | [2.041s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1306611Z  | [2.041s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'mvc', run:
2023-02-14T17:40:24.1306633Z  | [2.042s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list mvc
2023-02-14T17:40:24.1306750Z  | [2.042s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.1306783Z  | [2.042s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search mvc
2023-02-14T17:40:24.1306812Z  | [2.043s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1306840Z  | [2.044s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1308004Z  | [2.044s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.1308080Z  | [2.049s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.1308197Z  | [2.049s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: ==> /datadisks/disk1/work/A6F508D4/p/dotnet-cli/dotnet new "react" --debug:disable-sdk-templates --debug:custom-hive "/datadisks/disk1/work/A6F508D4/w/A5900923/e/Hives/.templateEngine" [/datadisks/disk1/work/A6F508D4/w/A5900923/e/]
2023-02-14T17:40:24.1309370Z  | [2.270s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'react'.
2023-02-14T17:40:24.1309541Z  | [2.273s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] Did you mean one of the following subcommands?
2023-02-14T17:40:24.1310556Z  | [2.274s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new create
2023-02-14T17:40:24.1310601Z  | [2.274s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search
2023-02-14T17:40:24.1310631Z  | [2.274s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1311482Z  | [2.275s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'react', run:
2023-02-14T17:40:24.1311541Z  | [2.275s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list react
2023-02-14T17:40:24.1311572Z  | [2.275s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.1311599Z  | [2.275s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search react
2023-02-14T17:40:24.1311625Z  | [2.275s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1311653Z  | [2.277s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1312696Z  | [2.277s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.1313571Z  | [2.282s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.1313620Z  | [2.283s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: ==> /datadisks/disk1/work/A6F508D4/p/dotnet-cli/dotnet new "reactredux" --debug:disable-sdk-templates --debug:custom-hive "/datadisks/disk1/work/A6F508D4/w/A5900923/e/Hives/.templateEngine" [/datadisks/disk1/work/A6F508D4/w/A5900923/e/]
2023-02-14T17:40:24.1313663Z  | [3.247s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.1313684Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'reactredux'.
2023-02-14T17:40:24.1313708Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1313736Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'reactredux', run:
2023-02-14T17:40:24.1313833Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list reactredux
2023-02-14T17:40:24.1314773Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.1315814Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search reactredux
2023-02-14T17:40:24.1315849Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1315880Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.1316708Z  | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.1317508Z  | [3.254s] Templates.Blazor.Test.BlazorWasmTemplateTest Error: Test threw an exception.
2023-02-14T17:40:24.1317547Z  | System.InvalidOperationException: Failed to uninstall previous templates. The template 'reactredux' could still be found.
2023-02-14T17:40:24.1317570Z  |    at Templates.Test.Helpers.TemplatePackageInstaller.VerifyCannotFindTemplateAsync(ITestOutputHelper output, String templateName) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 151
2023-02-14T17:40:24.1317698Z  |    at Templates.Test.Helpers.TemplatePackageInstaller.InstallTemplatePackages(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 114
2023-02-14T17:40:24.1317723Z  |    at Templates.Test.Helpers.TemplatePackageInstaller.EnsureTemplatingEngineInitializedAsync(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 64
2023-02-14T17:40:24.1317745Z  |    at Templates.Test.Helpers.ProjectFactoryFixture.CreateProject(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/ProjectFactoryFixture.cs:line 29
2023-02-14T17:40:24.1317766Z  |    at Templates.Blazor.Test.BlazorTemplateTest.CreateBuildPublishAsync(String auth, String[] args, String targetFramework, Boolean serverProject, Boolean onlyCreate) in /_/src/ProjectTemplates/Shared/BlazorTemplateTest.cs:line 49
2023-02-14T17:40:24.1317814Z  |    at Templates.Blazor.Test.BlazorWasmTemplateTest.BlazorWasmHostedPwaTemplateNoHttpsCanCreateBuildPublish() in /_/src/ProjectTemplates/test/Templates.Blazor.WebAssembly.Tests/BlazorWasmTemplateTest.cs:line 84
2023-02-14T17:40:24.1318012Z  |    at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_0.<<InvokeTestMethodAsync>b__1>d.MoveNext() in /_/src/xunit.execution/Sdk/Frameworks/Runners/TestInvoker.cs:line 264
2023-02-14T17:40:24.1318052Z  | --- End of stack trace from previous location ---
2023-02-14T17:40:24.1318078Z  |    at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in /_/src/xunit.execution/Sdk/Frameworks/ExecutionTimer.cs:line 48
2023-02-14T17:40:24.1318104Z  |    at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in /_/src/xunit.core/Sdk/ExceptionAggregator.cs:line 90
2023-02-14T17:40:24.1318130Z  | [3.266s] TestLifetime Information: Finished test BlazorWasmHostedPwaTemplateNoHttpsCanCreateBuildPublish in 3.258927s
2023-02-14T17:40:24.1318155Z 
2023-02-14T17:40:24.1318181Z 
```

--- a/src/Shared/Process/ProcessEx.cs
+++ b/src/Shared/Process/ProcessEx.cs
@@ -179,6 +179,7 @@ internal sealed class ProcessEx : IDisposable
                 _output.WriteLine("Process exited.");
             }
         }
+        // Don't remove this line - There is a race condition where the process exits and we grab the output before the stdout/stderr completed writing.
         _process.WaitForExit();
         _stdoutLines.CompleteAdding();
         _stdoutLines = null;

--- a/src/Shared/Process/ProcessEx.cs
+++ b/src/Shared/Process/ProcessEx.cs
@@ -179,7 +179,7 @@ internal sealed class ProcessEx : IDisposable
                 _output.WriteLine("Process exited.");
             }
         }
-
+        _process.WaitForExit();
         _stdoutLines.CompleteAdding();
         _stdoutLines = null;
         _exited.TrySetResult(_process.ExitCode);

